minard/HLDQTools.py

```diff
@@ -52,68 +52,86 @@ def generateHLDQProcStatus(ratdbDict):
-                return runNumber, -1, -1 
-    if data==None:
-        return runNumber, -1, -1 
-    
-    checkDict = {}
-    checkDict["fibre"] = data["fibre"]
-    checkDict["pulse_delay"] = data["pulse_delay"]
-    checkDict["avg_nhit"] = data["avg_nhit"]
-    checkDict["peak_amplitude"] = data["peak_amplitude"]
-    checkDict["max_nhit"] = data["max_nhit"]
-    checkDict["trigger"] = data["trigger"]
-    checkDict["run_length"] = data["run_length"]
-    checkDict["peak_number"] = data["peak_number"]
-    checkDict["prompt_time"] = data["prompt_time"]
-    checkDict["peak_time"] = data["peak_time"]
 
-    #Get the runinformation from the tellie dq output
-    runInformation = {}
-    runInformation["expected_tellie_events"] = data["check_params"]["expected_tellie_events"]
-    runInformation["actual_tellie_events"] = data["check_params"]["actual_tellie_events"]
-    runInformation["average_nhit"] = data["check_params"]["average_nhit"]
-    runInformation["greaterThanMaxNHitEvents"] = data["check_params"]["more_max_nhit_events"]
-    runInformation ["fibre_firing"] = data["check_params"]["fibre_firing"]
-    runInformation["fibre_firing_guess"] = data["check_params"]["fibre_firing_guess"]
-    runInformation["peak_number"] = data["check_params"]["peak_numbers"]
-    runInformation["prompt_peak_adc_count"] = data["check_params"]["prompt_peak_adc_count"]
-    runInformation["pre_peak_adc_count"] = data["check_params"]["pre_peak_adc_count"]
-    runInformation["late_peak_adc_count"] = data["check_params"]["late_peak_adc_count"]
-    runInformation["subrun_run_times"] = data["check_params"]["subrun_run_times"]
-    runInformation["pulse_delay_correct_proportion"]  = data["check_params"]["pulse_delay_efficiency"]
-
-    #Run Information for the subruns
-    runInformation["subrun_numbers"] = data["check_params"]["subrun_numbers"]
-    runInformation["avg_nhit_check_subruns"] = data["check_params"]["avg_nhit_check"]
-    runInformation["max_nhit_check_subruns"] = data["check_params"]["max_nhit_check"]
-    runInformation["peak_number_check_subruns"] = data["check_params"]["peak_number_check"]
-    runInformation["prompt_peak_amplitude_check_subruns"] = data["check_params"]["prompt_peak_amplitude_check"]
-    runInformation["prompt_peak_adc_count_check_subruns"] = data["check_params"]["prompt_peak_adc_count_check"]
-    runInformation["adc_peak_time_spacing_check_subruns"] = data["check_params"]["adc_peak_time_spacing_check"]
-    runInformation["pulse_delay_efficiency_check_subruns"] = data["check_params"]["pulse_delay_efficiency_check"]
-    runInformation["subrun_run_length_check"] = data["check_params"]["subrun_run_length_check"]
-    runInformation["correct_fibre_check_subruns"] = data["check_params"]["correct_fibre_check"]
-    runInformation["trigger_check_subruns"] = data["check_params"]["trigger_check"]
+    if data == {}:
```

Comment:
This probably works, but I think you can achieve the same (and a little easier to read) by doing:

```python
if data:
    return runs, -1, -1
```

or

```python
if len(data) == 0:
    return runs, -1, -1
```